### PR TITLE
[WIP][13.0][IMP] shopfloor: new scenario 'manual product transfer'

### PR DESCRIPTION
**Pull request by sebalix**
New scenario allowing to move a product or a lot from a source location to another one (similar to the "location content transfer" but limited to a specific product/lot).

Requires:
- https://github.com/OCA/stock-logistics-workflow/pull/851
- https://github.com/odoo/odoo/pull/71726 (required by `stock_restrict_lot`)

TODO:
- [x] Backend
- [x] Frontend
- [ ] Put in a separate module

Ref. 2152

### CODE REVIEW

File: shopfloor/services/manual_product_transfer.py [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+        Transitions:
+        * scan_product: to scan a product or a lot stored in the scanned location
+        * start: no stock found or wrong barcode
+        """
+        search = self._actions_for("search")
+        stock = self._actions_for("stock")
+        location = search.location_from_scan(barcode)
+        if not location:
+            return self._response_for_start(message=self.msg_store.no_location_found())
+        if not stock.is_src_location_valid(self, location):
```
Comment (TDu) [inline shopfloor/services/manual_product_transfer.py]:
```suggestion
        if not self.is_src_location_valid(location):
```
Maybe ?

File: shopfloor/services/manual_product_transfer.py [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+        return (move_lines - lines_other_picking_types, unreserved_moves, None)
+
+    def scan_destination_location(self, move_line_ids, barcode):
+        """Scan the destination location and post the move.
+
+        Transitions:
+        * start: move has been posted successfully
+        * scan_destination_location: scanned location is wrong (error)
+        """
+        move_lines = self.env["stock.move.line"].browse(move_line_ids).exists()
```
Comment (TDu) [inline shopfloor/services/manual_product_transfer.py:520]: Should we not check that the location bar code scanned is the correct one ?

Comment (sebalix) [reply shopfloor/services/manual_product_transfer.py:520]: Hey, but you are right! :D I totally missed this part...

Comment (sebalix) [reply shopfloor/services/manual_product_transfer.py:520]: Fixed

**Review by TDu — APPROVED** [review]

**Comment by mmequignon** [comment]
travis is :apple:

**Comment by jbaudoux** [comment]
obsolete

**Pull request closed without merging**